skhep/math/vectors.py

```diff
@@ -306,3 +308,236 @@ def __repr__(self):
+
+    def theta(self, deg=False):
+        """Return the spherical coordinate theta of the 3D-vector.
+
+        Options:
+           deg : return the angle in degrees (default is radians)
+        """
+        raise NotImplementedError
+
+    def phi(self, deg=False):
+        """Return the spherical or cylindrical coordinate phi of the 3D-vector.
+
+        Options:
+           deg : return the angle in degrees (default is radians)
+        """
+        raise NotImplementedError
+
+    @property
+    def px(self):
+        """Return the 3D-vector coordinate px, aka first momentum coordinate at position 0."""
+        return self.x
+
+    @px.setter
+    def px(self, value):
+        """Sets px, aka first momentum coordinate at position 0."""
+        self.__values[0] = value
+
+    @property
+    def py(self):
+        """Return the 3D-vector coordinate px, aka second momentum coordinate at position 1."""
+        return self.y
+
+    @py.setter
+    def py(self, value):
+        """Sets py, aka second momentum coordinate at position 1."""
+        self.__values[1] = value
+
+    @property
+    def pz(self):
+        """Return the 3D-vector coordinate pz, aka third momentum coordinate at position 2."""
+        return self.z
+
+    @pz.setter
+    def pz(self, value):
+        """Sets pz, aka third momentum coordinate at position 2."""
+        self.__values[2] = value
+
+    @property
+    def e(self):
+        """Return the energy/time component, aka momentum coordinate at position 3."""
+        return self.t
+
+    @e.setter
+    def e(self, value):
+        """Sets e, aka momentum coordinate at position 3."""
+        self.__values[3] = value
+
+    @property
+    def m(self):
+        """Return the invariant mass."""
+        return self.mag
+
+    @property
+    def m2(self):
+        """Return the square of the invariant mass."""
+        return self.mag2
+
+    @property
+    def mass(self):
+        """Return the invariant mass."""
+        return self.m
+
+    @property
+    def mass2(self):
+        """Return the square of the invariant mass."""
+        return self.m2
+
+    def transversemass(self):
+        """Return the transverse mass."""
+        raise NotImplementedError
+
+    def beta(self):
+        """Return :math:`\\beta = v/c`."""
+        raise NotImplementedError
+
+    def gamma(self):
+        """Return :math:`\\gamma = 1/\\sqrt{1-\\beta^2}`."""
+        raise NotImplementedError
+
+    def eta(self):
+        """Return the pseudorapidity."""
+        raise NotImplementedError
+
+    def pseudorapidity(self):
+        """"Return the pseudorapidity. Alternative to eta() method."""
+        return self.eta()
+
+    def rapidity(self):
+        """Return the rapidity."""
+        raise NotImplementedError
+
+    @property
+    def mag(self):
+        """Magnitude, a.k.a. norm, of the Lorentz vector."""
+        mag2 = self.mag2
+        return ( sqrt(mag2) if mag2 >= 0. else -sqrt(-mag2) )
```

Comment:
Ah, forgot that! My brain is not PEP8-compliant yet ;-). Will remove those parentheses.